packages/babel-preset-typescript/test/normalize-options.spec.js

```diff
@@ -0,0 +1,93 @@
+import normalizeOptions from "../src/normalize-options";
+describe("normalize options", () => {
+  (process.env.BABEL_8_BREAKING ? describe : describe.skip)("Babel 8", () => {
+    it("should throw on unknown options", () => {
+      expect(() => normalizeOptions({ allowNamespace: true })).toThrowError(
+        "@babel/preset-typescript: 'allowNamespace' is not a valid top-level option.\n- Did you mean 'allowNamespaces'?",
+      );
+    });
+    it.each([
+      "allowDeclareFields",
+      "allExtensions",
+      "allowNamespaces",
+      "isTSX",
+      "onlyRemoveTypeImports",
+    ])("should throw when `%p` is not a boolean", optionName => {
+      expect(() =>
+        normalizeOptions({ [optionName]: 0 }),
+      ).toThrowErrorMatchingSnapshot();
```

Comment:
Maybe we could just do this?

```suggestion
        ).toThrow(`@babel/preset-typescript: '${optionName}' option must be a boolean.`);
```

I don't think a snapshot really gives value here, since it's just a single line that could easily fit inline.